owa/app/js/components/view_log_entries/viewLogEntries.jsx

```diff
@@ -9,127 +9,308 @@
 
 import React from 'react';
 import ReactTable from 'react-table';
+import moment from 'moment';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
 import 'react-table/react-table.css';
 import {FaSearch} from 'react-icons/lib/fa';
-import {Input} from 'reactstrap'
+import {Input} from 'reactstrap';
+import apiCall from '../../utilities/apiHelper';
 
 export default class LogEntries extends React.Component {
-    constructor() {
-        super();
-        this.data = []
-        this.state = {
-            isHidden: true
-        }
+  constructor() {
+    super();
+    this.state = {
+      logEntries: [],
+      identifierSources:[],
+      isHidden: true,
+      filteredLogEntries: '',
+      startDate: '',
+      endDate: moment(new Date()),
+      dateChanged: false,
+      generatedByValue:'',
+      generatedBy:'',
+      searchFilters: {
+        source:'',
+        identifier: '',
+        comment: '',
+      }
+    };
+    this.getLogEntries = this.getLogEntries.bind(this);
+    this.getIdentifierSources = this.getIdentifierSources.bind(this);
+    this.advancedSearchView = this.advancedSearchView.bind(this);
+    this.advancedSearch = this.advancedSearch.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleChangeStart = this.handleChangeStart.bind(this);
+    this.handleChangeEnd = this.handleChangeEnd.bind(this);
+    this.resetSearchForm=this.resetSearchForm.bind(this);
+  }
+
+  getLogEntries() {
+    apiCall(null, 'get','/idgen/logentry?v=full').then((
+      response) => {this.setState({logEntries: response.results});
+    });
+  }
+
+  getIdentifierSources(){
+     apiCall(null, 'get', '/idgen/identifiersource?v=full').then((response) => {
+      this.setState({identifierSources: response.results});
+    });
+  }
 
+  advancedSearchView() {
+    this.setState({
+      isHidden: !this.state.isHidden
+    })
+  }
+
```

Comment:
Since you are updating the 'isHidden' state using a modification of its current value, you might want to consider using  the "prevState" construct. This will avoid any potential race-conditions that arise as a result of the setState method being an asynchronous one.